test/core/tsi/spiffe_ssl_transport_security_test.cc

```diff
@@ -0,0 +1,438 @@
+
+// Valid SPIFFE Bundles on both sides with the root configured for the
+// appropriate trust domain
+TEST_P(SpiffeSslTransportSecurityTest, MTLSSpiffe) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerKeyPath, kServerCertPath, kClientKeyPath, kClientCertPath,
+      kServerSpiffeBundleMapPath, kClientSpiffeBundleMapPath, kNonSpiffeCAPath,
+      /*expect_server_success=*/true,
+      /*expect_client_success_1_2=*/true, /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Valid SPIFFE Bundles on both sides with the root configured for the
+// appropriate trust domain, and a certificate chain with an intermediate CA on
+// the server side
+TEST_P(SpiffeSslTransportSecurityTest, MTLSSpiffeChain) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerChainKeyPath, kServerChainCertPath, kClientKeyPath,
+      kClientCertPath, kServerSpiffeBundleMapPath, kClientSpiffeBundleMapPath,
+      kNonSpiffeCAPath,
+      /*expect_server_success=*/true,
+      /*expect_client_success_1_2=*/true,
+      /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Just Spiffe bundles on the client side - the server side has the root
+// configured as a certificate
+TEST_P(SpiffeSslTransportSecurityTest, ClientSideSpiffeBundle) {
+  auto* fixture = new SslTsiTestFixture(kServerKeyPath, kServerCertPath,
+                                        kClientKeyPath, kClientCertPath, "",
+                                        kClientSpiffeBundleMapPath, kCaPemPath,
+                                        /*expect_server_success=*/true,
+                                        /*expect_client_success_1_2=*/true,
+                                        /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Just SPIFFE bundles on the server side - the client side has the root
+// configured as a certificate
+TEST_P(SpiffeSslTransportSecurityTest, ServerSideSpiffeBundle) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerKeyPath, kServerCertPath, kClientKeyPath, kClientCertPath,
+      kServerSpiffeBundleMapPath, "", kCaPemPath,
+      /*expect_server_success=*/true,
+      /*expect_client_success_1_2=*/true,
+      /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Use the good client spiffebundle on the server side so we don't get a
+// matching trust domain on the server side for the client's leaf certificate
+// Client-side TLS 1.3 sees success
+TEST_P(SpiffeSslTransportSecurityTest, MTLSSpiffeServerMismatchFail) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerKeyPath, kServerCertPath, kClientKeyPath, kClientCertPath,
+      kClientSpiffeBundleMapPath, kClientSpiffeBundleMapPath, kNonSpiffeCAPath,
+      /*expect_server_success=*/false,
+      /*expect_client_success_1_2=*/false,
+      /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Use the good server side spiffebundle on the client side so we don't get a
```

Comment:
Suggested edit (for consistency with the other comments above):

"Valid SPIFFE bundle on the server side, but the client side has a SPIFFE bundle that does not have a trust domain that will match the server leaf certificate."